api/db/migrations/20230307144120_add-locale-column-to-users-table.js

```diff
@@ -0,0 +1,18 @@
+const TABLE_NAME = 'users';
+const COLUMN_NAME = 'locale';
+
+exports.up = async function (knex) {
+  await knex.schema.table(TABLE_NAME, function (table) {
+    table.string(COLUMN_NAME).defaultTo(null);
+  });
+
+  return knex.raw(
+    `ALTER TABLE "users" ADD CONSTRAINT "users_locale_check" CHECK ( "locale" IN ('fr', 'en', 'fr-FR', 'fr-BE') )`
+  );
```

Comment:
juste pour être sur de bien comprendre, ici on retourne la table `user` en y ajoutant une contrainte qui est que dans la colonne `locale` on ne peut avoir que les valeurs : 'fr', 'en', 'fr-FR', 'fr-BE' c'est bien ça ?

Reply:
`.raw` veut dire que l'on va fournir à Knex une commande SQL standard. Cette commande va être exécutée, et c'est le résultat de cette exécution que l'on retourne (la promise).

Comme c'est une commande qui modifie la structure de la table, tu recevras surement un objet (ou autre) pour dire que la commande c'est bien exécuté en base de données.